balancer.go

```diff
@@ -0,0 +1,329 @@
+package xlb
+
+import (
+	"context"
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"github.com/google/uuid"
+	"github.com/rs/zerolog"
+	"github.com/xdire/xlb/tlsutil"
+	"net"
+	"os"
+	"strings"
+	"sync"
+	"time"
+)
+
+type ServicePool interface {
```

Comment:
It feels like this interface exists purely to abstract over `ServicePoolConfig`. Unless I'm missing something, this just seems to be additional complexity.

Reply:
Yes, at this moment it feels that the struct should be straight interface into balancer and there no reason for the interface.